pkg/controller/node/controller_utils.go

```diff
@@ -77,9 +86,31 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		}
 		remaining = true
 	}
+
+	if len(updateErrList) > 0 {
+		return false, utilerrors.NewAggregate(updateErrList)
+	}
 	return remaining, nil
 }
 
+// setPodTerminationReason attempts to set a reason and message in the pod status, updates it in the apiserver,
+// and returns an error if it encounters one.
+func setPodTerminationReason(kubeClient clientset.Interface, pod *api.Pod, nodeName string) (*api.Pod, error) {
+	if pod.Status.Reason == node.NodeUnreachablePodReason {
+		return pod, nil
+	}
+
+	pod.Status.Reason = node.NodeUnreachablePodReason
+	pod.Status.Message = fmt.Sprintf(node.NodeUnreachablePodMessage, nodeName, pod.Name)
+
+	var updatedPod *api.Pod
+	var err error
+	if updatedPod, err = kubeClient.Core().Pods(pod.Namespace).UpdateStatus(pod); err != nil {
```

Comment:
@gmarek 
The deletion is happening on the pod name itself without any precondition, so it cannot fail unless the pod is gone already. 

@deads2k, If any of the updates return an error, we keep track of that and reschedule the entire deletePods operation immediately after the execution and retry just setting status till we succeed. We don't stop the deletions because of it but we do retry.